go-server/go/api_default.go

```diff
@@ -0,0 +1,108 @@
+/*
+ * Web Server
+ *
+ * A simple web server for managing users
+ *
+ * API version: 1.0
+ * Contact: dmanor@redhat.com
+ * Generated by: Swagger Codegen (https://github.com/swagger-api/swagger-codegen.git)
+ */
+package swagger
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"path"
+)
+
+type userId int
+
+func AddUser(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	w.WriteHeader(http.StatusOK)
+
+	err := r.ParseForm()
+
+	if err != nil {
+		log.Fatal(err)
+	}
+	name, email := r.Form.Get("name"), r.Form.Get("email")
+	db := connectDB("localhost", 5432, "dmanor", "D1234", "webserverdb")
+	defer closeDB(db)
+
+	_, err = db.Exec("INSERT INTO users (name,email) VALUES ($1,$2);", name, email)
+	if err != nil {
+		return
+	}
+
+	var userId string
+	db.QueryRow("SELECT userid FROM users WHERE email = $1;", email).Scan(&userId)
+	resp, err := json.Marshal(userId)
+	if err != nil {
+		log.Fatal(err)
+	}
+	w.Write(resp)
+}
+
+func DeleteUser(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+
+	db := connectDB("localhost", 5432, "dmanor", "D1234", "webserverdb")
+	defer closeDB(db)
+
+	userId := path.Base(r.URL.Path)
+
+	res, err := db.Exec("DELETE FROM users WHERE userid=$1", userId)
+
+	if err != nil {
+		log.Fatal(err)
+	} else if val, _ := res.RowsAffected(); val == 0 {
+		fmt.Println(val, err)
+		w.WriteHeader(http.StatusConflict)
+		return
+	} else {
+		w.WriteHeader(http.StatusOK)
+	}
+}
+
+func FindUsers(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	w.WriteHeader(http.StatusOK)
+
+	db := connectDB("localhost", 5432, "dmanor", "D1234", "webserverdb")
+	defer closeDB(db)
+
+	rows, err := db.Query("SELECT * FROM users")
```

Comment:
What would happen if we have a lot of registered users (e.g. over 1,000,000,000 users)?